rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java

```diff
@@ -181,17 +187,17 @@ public static boolean isLockTx(BtcTransaction tx, Federation federation, Context
         return isLockTx(tx, Arrays.asList(federation), btcContext, bridgeConstants);
     }
 
-    public static boolean isReleaseTx(BtcTransaction tx, Federation federation, BridgeConstants bridgeConstants) {
-        int i = 0;
-        for (TransactionInput transactionInput : tx.getInputs()) {
-            try {
-                transactionInput.getScriptSig().correctlySpends(tx, i, federation.getP2SHScript(), Script.ALL_VERIFY_FLAGS);
-                // There is an input spending from the federation address, this is a release tx
+    private static boolean isReleaseTx(BtcTransaction tx, Federation federation) {
+        return isReleaseTx(tx, Collections.singletonList(federation));
+    }
+
+    public static boolean isReleaseTx(BtcTransaction tx, List<Federation> federations) {
```

Comment:
Now that we have this, it would be nice to use it within `isLockTx` (same behavior I believe).

Reply:
Yes, you are right, I'll extract it to a common method, thanks

Reply:
I'm not sure what you mean. I just meant replacing:
```
for (int i = 0; i < tx.getInputs().size(); i++) {
             final int index = i;
             if (federations.stream().anyMatch(federation -> scriptCorrectlySpendsTx(tx, index, federation.getP2SHScript()))) {
                 return false;
             }
         }
```
with
```
if (isReleaseTx(tx, federations)) {
  return false;
}
```

Reply:
yes, I would probably have figured that out once that _common method_ extraction turned to be just an unnecessary indirection. I'll implement it exactly as you said, thanks again